### PR TITLE
tests: add dbus interface spread test

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: tests/main/interfaces-dbus/task.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+environment:
+    DISPLAY: :0
+
+prepare: |
+    echo "Give a snap declaring a dbus slot in installed"
+    snap install --edge test-snapd-dbus-provider
+
+    echo "And the D-bus X11 dependencies are installed"
+    apt install -y dbus-x11
```
Comment (mvo5) [inline tests/main/interfaces-dbus/task.yaml]: Looks like we need to limit this to classic (via system:) if we apt here.

Comment (fgimenez) [inline tests/main/interfaces-dbus/task.yaml]: @mvo5 indeed, very good catch, thanks!

**Comment by niemeyer** [comment]
`/bin/bash: line 83: systemd-run: command not found` on 14.04.

**Comment by fgimenez** [comment]
@niemeyer thanks, i've added a specific upstart job for 14.04, please take another look.

**Comment by zyga** [comment]
@fgimenez I think you need to build and upload a ppc64el version of this snap:
```
+ snap install --edge test-snapd-dbus-provider
error: cannot install "test-snapd-dbus-provider": snap not found
```

**Comment by fgimenez** [comment]
@zyga thanks! the tests snaps are already in place

File: tests/main/interfaces-dbus/task.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    echo "And a snap declaring a matching dbus plug is installed"
+    snap install --edge test-snapd-dbus-consumer
+
+    echo "And the provider dbus loop is started"
+    dbus-launch > dbus.env
```
Comment (mvo5) [inline tests/main/interfaces-dbus/task.yaml]: The usual way to do this (as documented in the man-page) is ` eval $(dbus-launch --sh-syntax)` - is this not working for some reason?

Comment (fgimenez) [inline tests/main/interfaces-dbus/task.yaml]: @mvo5 exactly, when using `eval $(dbus-launch --sh-syntax)` I get:
```
+ sleep 1
+ MATCH 'hello world'
+ dbus-send --print-reply --dest=com.dbustest.HelloWorld /com/dbustest/HelloWorld com.dbustest.HelloWorld.SayHello
Failed to open connection to "session" message bus: /usr/bin/dbus-launch terminated abnormally with the following error: Autolaunch error: X11 initialization failed.

error: pattern not found, got:

+ sleep 1
+ MATCH 'hello world'
+ dbus-send --print-reply --dest=com.dbustest.HelloWorld /com/dbustest/HelloWorld com.dbustest.HelloWorld.SayHello
Failed to open connection to "session" message bus: /usr/bin/dbus-launch terminated abnormally with the following error: Autolaunch error: X11 initialization failed.
...
```

**Pull request merged**